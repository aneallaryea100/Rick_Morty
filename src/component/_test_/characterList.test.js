import { Provider } from 'react-redux';
import TestRenderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CharacterList from '../characterList';
import store from '../../redux/configureStore';

describe('CharacterList test', () => {
  test('snapshot', () => {
    const characters = [
      {
        id: 1,
        image: 'fge.jpeg',
        name: 'hekko',
      },
      {
        id: 2,
        image: 'fllge.jpeg',
        name: 'hekko',
      },
    ];
    const tree = TestRenderer.create(
      <BrowserRouter>
        <Provider store={store}>
          <CharacterList characters={characters} />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
