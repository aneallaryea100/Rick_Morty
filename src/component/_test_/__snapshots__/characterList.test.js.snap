// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList test snapshot 1`] = `
Array [
  <div
    className="cardMaincontainer"
  >
    <div>
      <div
        className="imageContainer"
      >
        <img
          alt="hekko"
          className="imageCharacterCard"
          src="fge.jpeg"
        />
      </div>
      <div
        className="cardFooter"
      >
        <p
          className="nameCharacter"
        >
          hekko
        </p>
        <button
          className="clickBtnCard"
          type="button"
        >
          <a
            className="detailedLink"
            href="/details/1"
            id={1}
            onClick={[Function]}
          >
            Details
          </a>
        </button>
      </div>
    </div>
  </div>,
  <div
    className="cardMaincontainer"
  >
    <div>
      <div
        className="imageContainer"
      >
        <img
          alt="hekko"
          className="imageCharacterCard"
          src="fllge.jpeg"
        />
      </div>
      <div
        className="cardFooter"
      >
        <p
          className="nameCharacter"
        >
          hekko
        </p>
        <button
          className="clickBtnCard"
          type="button"
        >
          <a
            className="detailedLink"
            href="/details/2"
            id={2}
            onClick={[Function]}
          >
            Details
          </a>
        </button>
      </div>
    </div>
  </div>,
]
`;
